File: C++/1512. NumberOfGoodPairs.cpp
/**
 * Solution to Number of Good Pairs in CPP
 *
 * author: ITES7321
 * ref:https://leetcode.com/problems/number-of-good-pairs/
 */

class Solution {
public:
  int numIdenticalPairs(vector<int>& nums) {
    sort(nums.begin(), nums.end());
    unordered_map<int, int> mp;
    int num_pairs = 0;
      
    for(int i = 0; i < nums.size(); i++) mp[nums[i]]++;
    for(auto x : mp) num_pairs+= x.second != 1 ? (int)(x.second * (x.second - 1) / 2) : 0;
      
    return num_pairs;
  }
};

File: Java/1512.NumberofGoodPairs.java
class Solution {
    public int numIdenticalPairs(int[] nums) {
        int cnt[] = new int[101], res = 0;
        for (var n: nums)
            ++cnt[n];
        for (int i = 0; i <= 100; ++i)
            res += cnt[i] * (cnt[i] - 1) / 2;
        return res;     
    }
}